proposals/csharp-7.0/tuples.md

```diff
@@ -62,164 +126,113 @@ tuple_type_element
+**ISSUE:** Where will the type System.Value be defined (and its public names listed)?
 
+The name given explicitly to any element shall not be the same as any name in the underlying type, except that an explicit name may have the form `Item`*N* provided it corresponds position-wise with an element of the same name in the underlying type.
+
+\[Example:
 ```csharp
-var t =  (ToString: 0, ObjectEquals: 1);  // error: names match underlying member names
-var t1 = (Item1: 0, Item2: 1);            // valid
-var t2 = (misc: 0, Item1: 1);             // error: "Item1" was used in a wrong position
+var t =  (ToString: 0, GetHashCode: 1);	// Error: names match underlying member names
+var t1 = (Item1: 0, Item2: 1);			// OK
+var t2 = (misc: 0, Item1: 1);			// Error: "Item1" used in a wrong position
 ```
+end example\]
 
-If the tuple is bigger than the limit of 7, the implementation will nest the "tail" as a tuple into the eighth element recursively. This nesting is visible by accessing the `Rest` field of a tuple, but that field my be hidden from e.g. auto-completion, just as the `ItemX` field names may be hidden but allowed when a tuple has named elements.
-
-A well formed "big tuple" will have names `Item1` etc. all the way up to the number of tuple elements, even though the underlying type doesn't physically have those fields directly defined. The same goes for the tuple returned from the `Rest` field, only with the numbers "shifted" appropriately.
+#### Element names and overloading, overriding, and hiding
 
-For tuple element names occurring in partial type declarations, the names must be the same.
+When tuple element names are used in overridden signatures or implementations of interface methods, tuple element names in parameter and return types shall be preserved. It is an error for the same generic interface to be inherited or implemented twice with identity-convertible type arguments that have conflicting tuple element names.
 
-```csharp
-partial class C : IEnumerable<(string name, int age)> { ... }
-partial class C : IEnumerable<(string fullname, int)> { ... } // error: names must be specified and the same
-```
+For the purpose of overloading, overriding and hiding, tuples of the same types and lengths as well as their underlying ValueTuple types shall be considered equivalent. All other differences are immaterial. When overriding a member it is permitted to use tuple types with the same names or names different than in the base member.
 
-When tuple elements names are used in overridden signatures, or implementations of interface methods, tuple element names in parameter and return types must be preserved. It is an error for the same generic interface to be inherited or implemented twice with identity convertible type arguments that have conflicting tuple element names
+**ISSUE:** Re the mention of "tuple length", which is not mentioned elsewhere; it seems to me that a tuple type includes an implied length based on the number of elements.
```

Comment:
That's how I interpret it. This is the same as the *arity* term. We should standardize on *arity*, and define it.

Reply:
Done!